judo-runtime-core-jsl-itest/src/test/java/hu/blackbelt/judo/runtime/core/jsl/entity/AbstractTest.java

```diff
@@ -0,0 +1,538 @@
+        assertEquals(iDao.getAll().size(),jDao.getAll().size());
+
+        //TODO JNG-4812
+//        iDao.delete(h.getCompositionIOnHSingleRequired()); //unfortunately you can delete required Component
+//        h = hDao.getById(h.identifier()).orElseThrow();
+
+        iDao.delete(h.getCompositionIOnHMulti().get(0));
+        assertTrue(iDao.getById(h.getCompositionIOnHMulti().get(0).identifier()).isEmpty());
+        h = hDao.getById(h.identifier()).orElseThrow();
+        assertEquals(2,h.getCompositionIOnHMulti().size());
+
+        //check the other methods are work well
+
+        //add
+        hDao.addCompositionIOnHMulti(h, Collections.singletonList(JtoI(jDao.create(J.builder().build()))));
+        assertEquals(2, h.getCompositionIOnHMulti().size());
+        h = hDao.getById(h.identifier()).orElseThrow();
+        assertEquals(3, h.getCompositionIOnHMulti().size());
+
+        assertEquals(3, hDao.countRelationIOnHMulti(h));
+        hDao.addRelationIOnHMulti(h, List.of(JtoI(jDao.create(J.builder().build()))));
+        assertEquals(4, hDao.countRelationIOnHMulti(h));
+        h = hDao.getById(h.identifier()).orElseThrow();
+
+        //remove
+        hDao.removeCompositionIOnHMulti(h, List.of(h.getCompositionIOnHMulti().get(0)));
+        assertEquals(3,h.getCompositionIOnHMulti().size());
+        h = hDao.getById(h.identifier()).orElseThrow();
+        assertEquals(2,h.getCompositionIOnHMulti().size());
+
+        assertEquals(4,hDao.countRelationIOnHMulti(h));
+        hDao.removeRelationIOnHMulti(h, List.of(hDao.queryRelationIOnHMulti(h).execute().get(0)) );
+        assertEquals(3,hDao.countRelationIOnHMulti(h));
+        h = hDao.getById(h.identifier()).orElseThrow();
+
+        hDao.removeCompositionIOnHMulti(h, List.of(h.getCompositionIOnHMulti().get(0),h.getCompositionIOnHMulti().get(1)));
+        assertEquals(2,h.getCompositionIOnHMulti().size());
+        h = hDao.getById(h.identifier()).orElseThrow();
+        assertEquals(0,h.getCompositionIOnHMulti().size());
+
+        assertEquals(3,hDao.countRelationIOnHMulti(h));
+        hDao.removeRelationIOnHMulti(h, List.of(hDao.queryRelationIOnHMulti(h).execute().get(0),hDao.queryRelationIOnHMulti(h).execute().get(1)) );
+        assertEquals(1,hDao.countRelationIOnHMulti(h));
+        h = hDao.getById(h.identifier()).orElseThrow();
+
+        //set
+        hDao.setCompositionIOnHSingle(h,JtoI(jDao.create(J.builder().withNameI("SetNewElement").build())));
+        hDao.setRelationIOnHSingle(h, JtoI(jDao.create(J.builder().withNameI("SetNewElement").build())));
+
+        h = hDao.getById(h.identifier()).orElseThrow();
+
+        assertEquals(Optional.of("SetNewElement"),h.getCompositionIOnHSingle().orElseThrow().getNameI());
+        assertEquals(Optional.of("SetNewElement"), hDao.queryRelationIOnHSingle(h).orElseThrow().getNameI());
+
+        //Check j and i instance ar equal because i is live with j only
```

Comment:
> Check j and i instance ar equal because i is live with j only

Check if instances of j and i are equal because all instances of i are also instances of j.